### ISSUE TITLE
Suggestion: Replace tuple return with struct using column name

### ISSUE THREAD

**Issue by Virgiel** [LouisGariepy/cornucopia#39]
For example, code like this:
```sql
--! example_query() *
SELECT Name, Country FROM Authors;
```

Should be equivalent to this:
```sql
--! example_query() {Name, Country} *
SELECT Name, Country FROM Authors;
```

Row column are always [named](https://docs.rs/postgres/latest/postgres/struct.Column.html). We should error on names clash and suggest using `AS`. We could also support names that are not valid rust identifiers with normalization or [raw identifier](https://doc.rust-lang.org/rust-by-example/compatibility/raw_identifiers.html).

How we should automatically handle nullable columns remains an open question. Maybe with something like this `{, ?, }`

What do you think of this?

**Comment by LouisGariepy** [comment]
Hi @Virgiel,

I think this would be a great plus for ergonomics. 

> We should error on names clash and suggest using AS...

Agreed!

> How we should automatically handle nullable columns remains an open question...

From what I've seen, some Cornucopia users are relying quite a bit on nullable columns, so I have to be careful with how I change them.

IMO, `{, ?, }` notation be a bit hard to read as you essentially have to count the number of commas.

What if we did something like this?
```sql
--! example_query_1() *
SELECT Name, Country FROM Authors;

--! example_query_2() {country} *
SELECT Name, Country FROM Authors;
```
This would generate structs like so
```rust
struct ExampleQuery1 {
    name: String,
    country: String
}

struct ExampleQuery2 {
    name: String,
    country: Option<String>
}
```
i.e. only columns whose names are found in `{...}` are marked as  nullable. I'm not yet set on anythin, its just a thought.

Please, let me know what you think :smile:

**Comment by LouisGariepy** [comment]
@Virgiel Query annotations like `--! example_query_2()?{country}?` would look a bit alien, but **If we go forward with the no-quantifiers design, I'd also go forward with this proposal, or at least something very similar** :+1: .

**Comment by LouisGariepy** [comment]
On track to be solved by #52.
